src/lib/biopam.ml

```diff
@@ -4,114 +4,153 @@ Provide tools via Biopam: https://github.com/solvuu/biopam
-  workflow_node cond ~name ~make ~edges
-
-let picard_tool ~host ~meta_playground =
-  Tool.create Tool.Definition.(biopam "picard")
-    ~ensure:(install_picard ~host ~meta_playground)
-
-(* A workflow to ensure that seq2HLA is available. *)
-let install_seq2HLA ~host ~meta_playground =
-  let open KEDSL in
-  let edges =
-    [ depends_on (configured ~host ~meta_playground ())
-    ; depends_on (Conda.configured ~host ~meta_playground)
-    ]
+  K.workflow_node cond ~name ~make ~edges
+
+let install_tool ?host ~install_path
+    ({package; test; edges; install_wrap; _ } as it) =
+  let edges = K.depends_on (configured ?host ~install_path ()) :: edges in
+  let name = "Installing " ^ package in
+  let make =
+    Opam.program_sh ~install_path "install %s" package
+    |> Option.value install_wrap ~default:(fun x -> x)
+    |> K.daemonize ?host
   in
-  let name  = "Installing seq2HLA" in
-  let opam fmt = Opam.com ~meta_playground fmt in
-  let make  =
-    daemonize ~host (Conda.run_in_biokepi_env ~meta_playground (Program.(sh (opam "install seq2HLA"))))
+  let path = Opam.which ~install_path it in
+  let test = (Option.value test ~default:default_test) ?host path in
+  let cond =
+    object
+      method is_done = Some (`Command_returns (test, 0))
+      method path = path
+    end
+  in
+  K.workflow_node cond ~name ~make ~edges
+
+let provide ?host ~install_path it =
+  let install_workflow = install_tool ?host ~install_path it in
+  let export_var, path =
+    match it.tool_type with
+    | Application -> "PATH", (Filename.dirname install_workflow#product#path)
+    | Library v   -> v, install_workflow#product#path
+  in
+  Tool.create Tool.Definition.(biopam it.package)
+    ~ensure:install_workflow
+    (* The 'FOO:+:' outputs ':' only if ${FOO} is defined. *)
+    ~init:(K.Program.shf "export %s=\"%s${%s:+:}${%s}\""
+              export_var path export_var export_var)
+
+let default ?host ~install_path () =
+  let mk ~package ~witness ?test ?(edges=[]) ?export_var ?install_wrap tt =
+    provide ?host ~install_path
+      { tool_type = tt ; package ; witness ; test ; edges ; install_wrap }
+  in
+  let version ?host path = K.Command.shell ?host (sprintf "%s --version" path) in
+  let need_conda =
+    [ K.depends_on (Conda.configured ?host ~install_path ())]
   in
-  let seq2HLA = Opam.file_command ~meta_playground ~package:"seq2HLA" `Bin "seq2HLA" in
-  let call_version = Command.shell ~host (seq2HLA ^ " --version") in
-  let cond = object method is_done = Some (`Command_returns (call_version, 0)) end in
-  workflow_node cond ~name ~make ~edges
-
-let seq2HLA_tool ~host ~meta_playground =
-  Tool.create Tool.Definition.(biopam "seq2HLA")
-    ~ensure:(install_seq2HLA ~host ~meta_playground)
-
-let toolkit ~host ~meta_playground () =
   Tool.Kit.create
-    [ picard_tool ~host ~meta_playground
-    ; seq2HLA_tool ~host ~meta_playground
+    [ mk (Library "PICARD_JAR") ~package:"picard"  ~witness:"picard.jar"
```

Comment:
(not in this PR, just noting to that github cross-links stuff)
we should keep in mind that there are now 2 ways to get Picard in Biokepi (with potentially different versions)
cf. https://github.com/hammerlab/biokepi/issues/34